md5.go

```diff
@@ -0,0 +1,51 @@
+package hashutil
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"errors"
+	"fmt"
+)
+
+type Md5 []byte
+
+var badMd5Length = fmt.Errorf("Md5 must have %d bytes length", md5.Size)
+
+// convert type Md5 to string representation (upper case)
+func (hash Md5) String() string {
+	return hex.EncodeToString(hash)
+}
+
+// StringToMd5 valid and convert string to Md5 type
+func StringToMd5(str string) (Md5, error) {
+	bytes, err := hex.DecodeString(str)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(bytes) != md5.Size {
+		return nil, badMd5Length
+	}
+
+	return bytes, nil
+}
+
+// BytesToMd5 valid and convert []bytes to Md5 type
+func BytesToMd5(bytes []byte) (Md5, error) {
+	if len(bytes) != md5.Size {
+		return nil, badMd5Length
+	}
+
+	for _, b := range bytes {
+		if b > 0xff {
```

Comment:
What's the point of this check? Can a byte be greater than `0xff`?

Reply:
no, this check don't sense - f1fc7353400748825ffd472fe534381a26871cdf

Reply:
Sorry, I forgot push this change... Now is fixed..